source/crud/tests/README.rst

```diff
@@ -90,10 +78,41 @@ Each YAML file has the following keys:
       - ``data``: The data that should exist in the collection after the
         operation has been run.
 
+Legacy Test Format for Single Operations
+----------------------------------------
+
+The test format above supports both multiple operations and APM expectations,
+and is consistent with the formats used by other specifications. Previously, the
+CRUD spec tests used a simplified format that only allowed for executing a
+single operation. Notable differences from the current format are as follows:
+
+- Instead of a `tests[i].operations` array, a single operation was defined as a
+  document in `tests[i].operation`. That document consisted of only the `name`,
+  `arguments`, and an optional `object` field.
+
+- Instead of `error` and `result` fields within each element in the
+  `tests[i].operations` array, the single operation's error and result were
+  defined under the `tests[i].outcome.error` and `tests[i].outcome.result`
+  fields.
+
+The legacy format should not conflict with the newer, multi-operation format
```

Comment:
I don't understand what is meant by this. @jmikola is this trying to say that test runners _should_ support both the updated and legacy formats? Do you intend to update all CRUD spec-tests to the new format or leave them as-is? IIUC whatever format we leave the CRUD tests in after this PR is the one drivers will _have_ to support (after they sync tests the next time).

Reply:
This is permitting drivers to support both. @xdg brought up a concern with this PR requiring more changes than necessary for drivers, which might distract from necessary 4.2 work. I do not intend to update all existing CRUD tests to this format.

My priority is to get the better, unified format documented so that you can use it for [SPEC-1153](https://jira.mongodb.org/browse/SPEC-1153), which requires APM. @kvwalker also needed `object:` syntax documented for [SPEC-1184](https://jira.mongodb.org/browse/SPEC-1184). At this point, I don't think @vincentkam really needed anything from this PR, as his retryable reads tests are consistent with the Transactions spec's format; however, the format in this PR should now be consistent with that as well.

Reply:
@prashantmital: I revised this paragraph a bit. Feel free to resolve if your original questions are answered.